vnets.tf

```diff
@@ -0,0 +1,55 @@
+#
+# This terraform plan defines the resources necessary to provision the Virtual
+# Networks in Azure according to IEP-002:
+#   <https://github.com/jenkins-infra/iep/tree/master/iep-002>
+#
+#                        +---------------------+
+#                        |                     |
+#      +---------------> |  Public Production  <-------+
+#      |                 |                     |       |
+#      |                 +---------------------+     VNet Peering
+#      |                                               |
+#      |                                 +-------------v--------+
+#                        +-------------+ |                      |
+# The Internet --------> + VPN Gateway |-|  Private Production  |
+#                        +-------------+ |                      |
+#      |                                 +----------------------+
+#      |
+#      |                 +----------------+
+#      |                 |                |
+#      +---------------> |   Development  |
+#                        |                |
+#                        +----------------+
+#
+## RESOURCE GROUPS
+################################################################################
+data "azurerm_resource_group" "public_prod" {
+  name = "prod-jenkins-public-prod"
```

Comment:
What's the purpose of the repetition of "prod"? How about "jenkins-public-prod"?

Reply:
It's a `data` in term of terraform, meaning that it is a reference to an existing resource. So we do not have the choice except using the actual name of the existing resource.

@smerle33 and I wondered the same and found the following explanations for this naming though:
- https://github.com/jenkins-infra/azure/blob/legacy-tf/plans/vnets.tf#L28 => there is a "prefix" variable,
- which has no default value: https://github.com/jenkins-infra/azure/blob/legacy-tf/plans/variables.tf#L13-L14.
- The value `prod` for this variable was defined by the pipeline (https://github.com/jenkins-infra/azure/blob/legacy-tf/Jenkinsfile#L26-L40) and passed to terraform through a variables file (https://github.com/jenkins-infra/azure/blob/legacy-tf/Jenkinsfile#L73)

=> We understood it was a way to separate prod / staging (named `infraci`) way before the terraform workdir. Difference with our current setup is that we define purely separated environments for prod and staging (better safe than sorry :D )